### PR TITLE
add libhtp-rs project

### PR DESCRIPTION
**Pull request by koltiradw**
Maybe I'm not quite brave for the whole Suricata :cold_sweat: , but I think we can start with libhtp-rs.

### CODE REVIEW

File: projects/libhtp-rs/sydr_and_cover/targets/sydr_htp.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        let mut f = File::open(filename).expect("no file found");
+        let metadata = std::fs::metadata(filename).expect("unable to read metadata");
+        let mut data = vec![0; metadata.len() as usize];
+        f.read(&mut data).expect("buffer overflow");
+        let mut t = Test::new(TestConfig());
+        t.run_slice(&data);
+    }
+    Ok(())
+}
+Footer
```
Comment (Avgor46) [inline projects/libhtp-rs/sydr_and_cover/targets/sydr_htp.rs:22]: Footer?

Comment (koltiradw) [reply projects/libhtp-rs/sydr_and_cover/targets/sydr_htp.rs:22]: ahahaa, sry))

File: projects/libhtp-rs/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Minimize corpus:
+
+    $ sydr-fuzz -c sydr-fuzz.toml cmin
+
+Check security predicates:
+
+    $ sydr-fuzz -c sydr-fuzz.toml security
+
+Get coverage report:
```
Comment (anfedotoff) [inline projects/libhtp-rs/README.md:39]: Please, use sydr-fuzz capabilities:
https://github.com/ispras/oss-sydr-fuzz/tree/master/projects/goblin#run-hybrid-fuzzing

File: projects/libhtp-rs/cover.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+mkdir /cover
```
Comment (anfedotoff) [inline projects/libhtp-rs/cover.sh:1]: We could do it this way :

    $ sydr-fuzz -c sydr-fuzz.toml cov-export -- -format=lcov > htp.lcov

Script is not needed

File: projects/libhtp-rs/sydr-fuzz-afl++.toml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+################################################################################
+exit_on_time = 7200
+
+[sydr]
+args = "-l debug"
+target = "/sydr_fuzz_htp_rs @@"
+
+[aflplusplus]
+args = "-i /corpus -t 100000 -x /http.dict"
+target = " /afl_fuzz_htp_rs @@"
```
Comment (anfedotoff) [inline projects/libhtp-rs/sydr-fuzz-afl++.toml:24]: [cov] section.
https://github.com/ispras/oss-sydr-fuzz/blob/master/projects/goblin/parse.toml#L24

File: projects/libhtp-rs/sydr-fuzz.toml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+################################################################################
+exit_on_time = 7200
+
+[sydr]
+args = "-l debug"
+target = "/sydr_fuzz_htp_rs @@"
+
+[libfuzzer]
+path = "/cargo_fuzz_htp_rs"
+args = "-rss_limit_mb=8192 -max_len=4096 -dict=/http.dict /corpus"
```
Comment (anfedotoff) [inline projects/libhtp-rs/sydr-fuzz.toml:24]: [cov] section.
https://github.com/ispras/oss-sydr-fuzz/blob/master/projects/goblin/parse.toml#L24

**Review by anfedotoff — CHANGES_REQUESTED** [review]

File: projects/libhtp-rs/cover.sh (hunk 3 above)
Comment (koltiradw) [reply projects/libhtp-rs/cover.sh]: i had some problems with llvm-profdata merge.
I hope it was just a mistake on my part.

File: projects/libhtp-rs/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+WORKDIR /
+
+COPY build.sh http.dict ./
+COPY afl/ /libhtp-rs/afl/
+COPY sydr_and_cover/ /libhtp-rs/sydr_and_cover/
+
+RUN bash build.sh
+
+ENV PATH=/root/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/:$PATH
```
Comment (anfedotoff) [inline projects/libhtp-rs/Dockerfile:37]: No new line at end of file

File: projects/libhtp-rs/htp-afl++.toml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+[sydr]
+args = "-l debug"
+target = "/sydr_fuzz_htp_rs @@"
+
+[aflplusplus]
+args = "-i /corpus -t 100000 -x /http.dict"
+target = " /afl_fuzz_htp_rs @@"
+
+[cov]
+target = "/cover_fuzz_htp_rs @@"
```
Comment (anfedotoff) [inline projects/libhtp-rs/htp-afl++.toml:27]: No new line at end of file

File: projects/libhtp-rs/htp.toml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+[sydr]
+args = "-l debug"
+target = "/sydr_fuzz_htp_rs @@"
+
+[libfuzzer]
+path = "/cargo_fuzz_htp_rs"
+args = "-rss_limit_mb=8192 -max_len=4096 -dict=/http.dict /corpus"
+
+[cov]
+target = "/cover_fuzz_htp_rs @@"
```
Comment (anfedotoff) [inline projects/libhtp-rs/htp.toml:27]: No new line at end of file

**Review by anfedotoff — APPROVED** [review]

**Pull request merged**